examples/packetgen/packetgen.lua

```diff
@@ -0,0 +1,1262 @@
+    "\n| Bytes received             : %32u |"                               ..
+    "\n| Byte receive rate          : %32.2f |"                             ..
+    "\n| Packets dropped on send    : %32u |"                               ..
+    "\n| Packets dropped on receive : %32u |"                               ..
+    "\n| TX packets short           : %32u |"                               ..
+    "\n| RX mean latency            : %32.10f |"                            ..
+    "\n| RX mean2 latency           : %32.10f |"                            ..
+    "\n+---------------------------------------------------------------+"   ,
+    coreid                                                                  ,
+    portid                                                                  ,
+    self.txPackets                                                          ,
+    (self.txPackets - self.prevTxPackets) / self.elapsedTime                ,
+    self.rxPackets                                                          ,
+    (self.rxPackets - self.prevRxPackets) / self.elapsedTime                ,
+    self.txBytes                                                            ,
+    (self.txBytes - self.prevTxBytes) / self.elapsedTime                    ,
+    self.rxBytes                                                            ,
+    (self.rxBytes - self.prevRxBytes) / self.elapsedTime		    ,
+    self.txDropped                                                          ,
+    self.rxDropped                                                          ,
+    self.txShort                                                            ,
+    self.rxMeanLatency                                                      ,
+    self.rxM2Latency
+  )
+  
+  print(statsString)
+end
+
+-- Utilities ------------------------------------------------------------------
+
+-- @brief Converts a number to a hex string representation.
+-- @param num The number to convert to hex.
+function num2hex(num)
+    local hexstr = '0123456789abcdef'
+    local s      = ''
+    while num > 0 do
+        local mod = math.fmod(num, 16)
+        s         = string.sub(hexstr, mod + 1, mod + 1) .. s
+        num       = math.floor(num / 16)
+    end
+    if s == '' then s = '0' end
+    return s
+end
+
+-- Converts a mac address in number format (a long) to a string based mac
+-- address of the form: a:b:c:d:e:f
+-- @param mac The mac address to convert
+function convertMacNumberToString(mac) 
+    local macArray = {}
+    for i = 6, 1, -1 do
+        macArray[i] = bit.band(mac, 0xff)
+        mac         = bit.rshift(mac, 8)
+    end
+    local macString = string.format("%u:%u:%u:%u:%u:%u",
+      macArray[1], macArray[2], macArray[3],
+      macArray[4], macArray[5], macArray[6]
+    )
+    return macString
+end
+
+-- @brief Generates a random number between 1 - 120 for the IMIX.
+function imixSize() 
+    math.randomseed(os.time())
```

Comment:
reseeding the PRNG for each call is kind of pointless -- even wrong in this case as os.time() has a granularity of 1 second